pkg/engine/api/types.go

```diff
@@ -29,10 +29,15 @@ type Raw struct {
 
 type Systemd struct {
 	TargetPath string `mapstructure:"targetPath"`
-	User       string `mapstructure:"user"`
-	Schedule   string `mapstructure:"schedule"`
-	InitialRun bool
-	LastCommit *object.Commit
+	Root       bool   `mapstructure:"root"`
+	// If non-root, home directory must be supplied to place unit file at $HOME/.config/systemd/user
```

Comment:
@sallyom should we do the equivalent of a mkdir -p just to ensure the path exists? I know its extra work on us but I dont think the subdirectory exists by default

Reply:
we may also have to do a chown as well or set permissions to be 777

Reply:
that would be just for the non root users

Reply:
I _think_ in order for homeDir to be non-root, host podman would be non-root socket, which means that directory will exist? I see `~/.config/systemd/user/sockets.target.wants/podman.socket` on my system and am assuming the act of `systemctl --user enable podman.socket` creates that if it doesn't exist. wdyt?

Reply:
Ah yes good call.